File: NoahCode/MongoCom/MongoCom/logInfo.cs
using System;
using MongoDB.Bson;
using MongoDB.Driver;
using System.Collections.Generic;
using System.Text;

namespace MongoCom
{
    public class LogInfo
    {
        private BsonDocument document;
        private String tourName;
        private String userID;
        private Double userHeight;
        private Double objLocX;
        private Double objLocY;
        private Double objLocZ;
        private Double roll;
        private Double pitch;
        private Double yaw;
        private Double w;
        private int time;
        private String stamp;

        public LogInfo(BsonDocument recievedDocument)
        {
            this.document = recievedDocument;
            this.tourName =  (string)recievedDocument["tourName"];
            this.userID = (string)recievedDocument["userID"];
            this.userHeight = Convert.ToDouble(recievedDocument["userHeight"]); 
            this.objLocX = Convert.ToDouble(recievedDocument["x-coordinate"]);
            this.objLocY = Convert.ToDouble(recievedDocument["y-coordinate"]);
            this.objLocZ = Convert.ToDouble(recievedDocument["z-coordinate"]);
            this.roll = Convert.ToDouble(recievedDocument["roll"]);
            this.pitch = Convert.ToDouble(recievedDocument["pitch"]);
            this.yaw = Convert.ToDouble(recievedDocument["yaw"]);
            this.w = Convert.ToDouble(recievedDocument["w"]);
            this.time = Convert.ToInt32(recievedDocument["time"]);
            this.stamp = (string)recievedDocument["timeStamp"].ToString();
            //Need to convert this one to string array and test this 
            //this.info = (BsonArray)recievedDocument["info"];
        }
        public BsonDocument returnDoc()
        {
            return document;
        }
       public string returnTourName()
        {
            return this.tourName;
        }
        public string returnUserID()
        {
            return this.userID;
        }
        public Double returnUserHeight()
        {
            return this.userHeight;
        }
        public Double returnX()
        {
            return this.objLocX;
        }
        public Double returnY()
        {
            return this.objLocY;
        }
        public Double returnZ()
        {
            return this.objLocZ;
        }
        public Double returnRoll()
        {
            return this.roll;
        }
        public Double returnPitch()
        {
            return this.pitch;
        }
        public Double returnYaw()
        {
            return this.yaw;
        }
        public Double returnW()
        {
            return this.w;
        }
        public int returnTime()
        {
            return this.time;
        }
        public String returnStamp()
        {
            return this.stamp;
        }
    }
}

File: Sandbox3/Sandbox/app/src/main/java/com/sandbox/sandbox/MongoCom/LogInfo.java
package com.sandbox.sandbox.MongoCom;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;

import org.bson.BsonDocument;
import org.bson.Document;
import org.bson.types.ObjectId;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

import static java.lang.System.in;
import static java.lang.System.nanoTime;

public class LogInfo
{
    private Document document;
    private String tourName;
    private String userID;
    private float userHeight;
    private float objLocX;
    private float objLocY;
    private float objLocZ;
    private float roll;
    private float pitch;
    private float yaw;
    private float w;
    private float stamp;

    public LogInfo(String name, String userid, float height, float x, float y, float z, float rx, float ry, float rz, float rw, float time)
    {
        this.tourName = name;
        this.userID = userid;
        this.userHeight = height;
        this.objLocX = x;
        this.objLocY = y;
        this.objLocZ = z;
        this.roll = rx;
        this.pitch = ry;
        this.yaw = rz;
        this.w = rw;
        this.stamp = time;
        document = new Document();
        document.put("tourName", name);
        document.put("userID", userid);
        document.put("objectType", "userLogData");
        document.put("userHeight", height);
        document.put("x-coordinate", x);
        document.put("y-coordinate", y);
        document.put("z-coordinate", z);
        document.put("roll", rx);
        document.put("pitch", ry);
        document.put("yaw", rz);
        document.put("w", rw);
        document.put("time", time);
        document.put("timeStamp", new Timestamp(System.currentTimeMillis()).toString());
    }
    public String returnUserId()
    {
        return userID;
    }
    public LogInfo(Document recievedDocument)
    {
        this.document = recievedDocument;
        this.tourName = recievedDocument.get("tourName").toString();
        this.userID = recievedDocument.get("userID").toString();
        this.objLocX = Float.valueOf(recievedDocument.get("x-coordinate").toString());
        this.objLocY = Float.valueOf(recievedDocument.get("y-coordinate").toString());
        this.objLocZ = Float.valueOf(recievedDocument.get("z-coordinate").toString());
        this.roll = Float.valueOf(recievedDocument.get("roll").toString());
        this.pitch = Float.valueOf(recievedDocument.get("pitch").toString());
        this.yaw = Float.valueOf(recievedDocument.get("yaw").toString());
        this.w = Float.valueOf(recievedDocument.get("w").toString());
        this.stamp = Integer.valueOf(recievedDocument.get("time").toString());
        //Need to convert this one to string array and test this
        //this.info = (BsonArray)recievedDocument["info"];
    }
    public Document returnDoc()
    {
        return document;
    }

}
